pkgs/by-name/ha/handheld-daemon/package.nix

```diff
@@ -1,63 +1,96 @@
 {
+  lib,
+  python3Packages,
   fetchFromGitHub,
+
+  # dependencies
+  systemd,
   hidapi,
+  coreutils,
   kmod,
-  lib,
-  python3,
-  toybox,
-  lsof
+  dbus,
+  lsof,
 }:
-python3.pkgs.buildPythonApplication rec {
+python3Packages.buildPythonApplication rec {
   pname = "handheld-daemon";
-  version = "3.6.1";
+  version = "3.6.2";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "hhd-dev";
     repo = "hhd";
     rev = "refs/tags/v${version}";
-    hash = "sha256-IdpSRb66G+WzTv/BC29r2OjO1b4VdWbV6OSzOoiFAO0=";
+    hash = "sha256-W1Ap6yTryBDozKe3aO413Fu0RBul9kEA9ACUTdYyOKM=";
   };
 
-  propagatedBuildInputs = with python3.pkgs; [
+  postPatch =
+    let
+      udevadm = lib.getExe' systemd "udevadm";
+    in
+    ''
+      # handheld-daemon contains a fork of the python module `hid`, so this hook
+      # is borrowed from the `hid` derivation.
+      hidapi=${lib.getLib hidapi}/lib/
+      test -d $hidapi || { echo "ERROR: $hidapi doesn't exist, please update/fix this build expression."; exit 1; }
```

Comment:
This test and comment are holdovers from the `hid` derivation.  Since you've changed so much, you can probably remove them here (and might want to open a PR for that package with similar changes to what you've done here).

Reply:
Yes, you are right. I at least removed it from here.